dotcom-rendering/scripts/webpack/gu-stats-report-plugin.js

```diff
@@ -0,0 +1,158 @@
+const fetch = require('node-fetch');
+const os = require('os');
+const { exec } = require('child_process');
+
+class GuStatsReportPlugin {
+	constructor(config) {
+		this.buildName = config?.buildName;
+		this.project = config?.project;
+		this.team = config?.team;
+		this.buildCount = 0;
+		this.gitBranch = null;
+		this.gitHash = null;
+		this.machineUsername = null;
+		this.machineHostname = os.hostname();
+		this.fetchMachineUsername();
+		this.fetchGitBranch();
+		this.fetchGitHash();
+		if (config.displayDisclaimer)
+			console.log(
+				'[gu-stats-report] This project reports build stats & basic machine information for internal use only. We will use this information to help us improve developer experience for this project.',
+			);
+	}
+
+	isValidConfig() {
+		if (!this.buildName || !this.project || !this.team) return false;
+		return true;
+	}
+
+	fetchMachineUsername() {
+		try {
+			// if 'username' is undefied os.userInfo() will throw an error
+			// https://nodejs.org/api/os.html#osuserinfooptions
+			this.machineUsername = os.userInfo().username;
+		} catch {
+			console.error('[gu-stats-report] Failed to get machine username');
+		}
+	}
+
+	fetchGitBranch() {
+		exec('git branch --show-current', (err, stdout) => {
+			if (err)
+				return console.error(
+					'[gu-stats-report] Failed to get current git branch',
+				);
+			this.gitBranch = stdout.trim();
+		});
+	}
+
+	fetchGitHash() {
+		exec('git rev-parse --short HEAD', (err, stdout) => {
+			if (err)
+				return console.error(
+					'[gu-stats-report] Failed to get current git hash',
+				);
+			this.gitHash = stdout.trim();
+		});
+	}
+
+	apply(compiler) {
+		const onDone = (stats) => {
+			// Increment the buildCount
+			this.buildCount += 1;
+
+			if (!this.isValidConfig)
+				return console.log(
+					'[gu-stats-report] Unable to report stats - invalid config',
+				);
+
+			const URL = 'https://logs.code.dev-guardianapis.com/log';
+			fetch(URL, {
+				method: 'POST',
+				body: JSON.stringify({
+					label: 'buildstats',
+					properties: [
+						{
+							name: 'project',
+							value: 'dotcom-rendering',
+						},
+						{
+							name: 'team',
+							value: 'dotcom',
+						},
+						{
+							name: 'machineUsername',
+							value: this.machineUsername || 'unknown',
+						},
+						{
+							name: 'machineHostname',
+							value: this.machineHostname,
+						},
```

Comment:
Should we be reporting anonymously? These are cool and allow us (among other things) to know how many unique 'uses' DCR has, as well as specifically find and work to diagnose anyone who might be having a bad experience.

However whether it's worth collecting 'identifying' information to achieve this goal should definitely be discussed

Reply:
Worth checking in with InfoSec what we would need to do this? Could also be worth checking with people who already use it how they feel.

If they're good - potentially always show the disclaimer and a way of opting out?

Reply:
Might be worth thinking of doing it anonymously first as we probably want to support that and add identification as a second PR?

Reply:
I've updated this PR for the time being to use uuid (v4 - we could perhaps look at using v1 which is based on time & mac address) to generate 'session ids'. These session IDs are consistent across all 3 builds so do give us the benefit of 'usage' statistics ~ and we can later look at using anonymous ID's (is MAC address a good candidate for this?) or other options.